### PR TITLE
xero import settings html coding

### PR DESCRIPTION
**Pull request by DhaaraniCIT**

### CODE REVIEW

**Comment by DhaaraniCIT** [comment]
Screenshot 2024-03-12 at 6 15 44 PM
Screenshot 2024-03-12 at 6 15 47 PM

File: src/app/branding/branding-config.ts [hunk 1]
```diff
@@ -295,12 +295,12 @@ const content: ContentConfiguration = {
                     contentText: '',
                     importCategoriesLabel: 'Import the Chart of Accounts as Categories in ',
                     importCategoriesSubLabel: 'Imported account will be available as Categories in ' + brandingConfig.brandName + '.',
-                    importItemsLabel: 'Import Products/Services from Xero',
-                    importItemsSubLabel: 'Products/services from Xero will be imported as Categories in ',
+                    importCustomersLabel: 'Import Customers from Xero',
+                    importCustomersSubLabel: 'The Customers in Xero will be imported as Projects in Fyle and will be a selectable field while creating an expense',
```
Comment (ashwin1111) [inline src/app/branding/branding-config.ts:299]: will confirm this tomo

File: src/app/branding/branding-config.ts [hunk 2]
```diff
@@ -295,12 +295,12 @@ const content: ContentConfiguration = {
...
                     taxCodeLabel: 'Import Tax from Xero',
                     taxCodeSubLabel: 'The imported Tax codes from Xero will be set as Tax group in ',
                     defaultTaxCodeLabel: 'Select Default Tax Code',
-                    importVendorsAsMerchantsLabel: 'Import Vendors from Xero'
+                    importSuppliersAsMerchantsLabel: 'Import Suppilers from Xero as Merchants'
```
Comment (ashwin1111) [inline src/app/branding/branding-config.ts:303]: typo, Suppliers

File: src/app/core/models/xero/xero-configuration/xero-import-settings.model.ts [hunk 3]
```diff
@@ -62,7 +63,18 @@ export class XeroImportSettingModel extends ImportSettingsModel {
   }
 
   static mapAPIResponseToFormGroup(importSettings: XeroImportSettingGet | null, xeroFields: IntegrationField[]): FormGroup {
-    const expenseFieldsArray = importSettings?.mapping_settings ? this.constructFormArray(importSettings.mapping_settings, xeroFields) : [];
+    let additionalOption: any[] = [];
+    if (brandingConfig.brandId === 'co') {
+      const coaddtional = {
+        source_field: 'XeroFyleField.PROJECT',
```
[the remark this answers is not in the document]
Comment (DhaaraniCIT) [reply src/app/core/models/xero/xero-configuration/xero-import-settings.model.ts:69]: that is like a placeholder, dummy one

File: src/app/shared/components/configuration/configuration-toggle-field/configuration-toggle-field.component.html [hunk 4]
```diff
@@ -22,6 +22,6 @@ <h5 class="!tw-text-text-muted tw-text-14-px !tw-font-400 !tw-leading-4" [ngClas
       </div>
     </div>
     <div *ngIf="!hideToggle" class="configuration--input-toggle-section">
-      <p-inputSwitch [formControlName]="formControllerName"></p-inputSwitch>
+      <p-inputSwitch [formControlName]="formControllerName" [disabled]="disabled" [tooltipPosition]="'top'" [pTooltip]="disabled ? 'You have already mapped a tracking category from Xero to the Project field in '+ brandingConfig.brandName +'. Change the configured mapping to a new field to be able to import Customers in the Project field.' : ''"></p-inputSwitch>
```
[the remark this answers is not in the document]
Comment (DhaaraniCIT) [reply src/app/shared/components/configuration/configuration-toggle-field/configuration-toggle-field.component.html:25]: we have only one condition here

File: src/app/integrations/xero/xero-shared/xero-import-settings/xero-import-settings.component.ts [hunk 5]
```diff
@@ -183,6 +191,42 @@ export class XeroImportSettingsComponent implements OnInit {
...
+            this.fyleExpenseFields.push({ attribute_type: XeroFyleField.PROJECT, display_name: 'Project', is_dependent: false });
+            this.fyleExpenseFields.push(this.customFieldOption[0]);
+          }
+        }
+      });
+    }
+  }
+
+  customerOptions() {
+    this.xeroExpenseFields.push({ attribute_type: XeroFyleField.CUSTOMER, display_name: 'Customer' });
```
[the remark this answers is not in the document]
Comment (DhaaraniCIT) [reply src/app/integrations/xero/xero-shared/xero-import-settings/xero-import-settings.component.ts:245]: customer is not coming as a option through the API 
Screenshot 2024-03-14 at 11 57 08 AM

File: src/app/integrations/xero/xero-shared/xero-import-settings/xero-import-settings.component.ts [hunk 6]
```diff
@@ -183,6 +191,42 @@ export class XeroImportSettingsComponent implements OnInit {
...
 
+  private createImportCustomerWatcher(): void {
+    if (brandingConfig.brandId === 'co') {
+      const formArray = this.importSettingsForm.get('expenseFields') as FormArray;
+      const index = formArray.value.findIndex((data:any) => data.destination_field === XeroFyleField.CUSTOMER);
+      formArray.controls.at(index)?.get('import_to_fyle')?.valueChanges.subscribe((isCustomerImportEnabled) => {
+        if (isCustomerImportEnabled) {
+          formArray.controls.at(index)?.get('source_field')?.patchValue(XeroFyleField.PROJECT);
+        } else {
+          formArray.controls.at(index)?.get('source_field')?.patchValue('XeroFyleField.PROJECT');
```
[the remark this answers is not in the document]
Comment (DhaaraniCIT) [reply src/app/integrations/xero/xero-shared/xero-import-settings/xero-import-settings.component.ts:224]: DISABLED_XERO_SOURCE_FIELD

**Review by ashwin1111 — APPROVED** [review]

File: src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.ts [hunk 7]
```diff
@@ -110,22 +110,31 @@ export class ConfigurationImportFieldComponent implements OnInit {
...
+      this.isXeroProjectMapped = false;
+      this.xeroProjectMapping.emit(this.isXeroProjectMapped);
     }
   }
 
+  getOptions(expenseField: AbstractControl): FyleField[]{
+    if(expenseField.value.destination_field === 'CUSTOMER' && this.appName === AppName.XERO && !expenseField.value.import_to_fyle) {
+      return this.filteredFyleFields
+    } else if (expenseField.value.source_field === 'CATEGORY') {
+      return this.fyleFieldOptions
```
[the remark this answers is not in the document]
Comment (DhaaraniCIT) [reply src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.ts]: if we use filteredFyleFields, `category` option will be removed
Screenshot 2024-03-15 at 12 53 47 PM

[the remark this answers is not in the document]
Comment (ashwin1111) [reply src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.ts]: yeah, that's expected, category option in dropdown should not be visible

**Pull request merged**